Lib/bdb.py

```
@@ -39,6 +214,11 @@ def __init__(self, skip=None):
         self.trace_opcodes = False
         self.enterframe = None
         self.code_linenos = weakref.WeakKeyDictionary()
+        self.backend = backend
+        if backend == 'monitoring':
+            self.monitoring_tracer = _MonitoringTracer()
+        else:
```

Maybe raise an exception if `backend != "settrace"`?